src/D2L.CodeStyle.Analyzers/Immutability/MutabilityAuditor.cs

```diff
@@ -3,15 +3,24 @@
 
 namespace D2L.CodeStyle.Analyzers.Immutability {
 	internal sealed class MutabilityAuditor {
-		public static bool IsAudited( ISymbol symbol, out Location location ) {
+		public static bool IsAudited( ISymbol symbol, out Location location, out Diagnostic diagnostic ) {

```

Comment:
As referred to in https://github.com/Brightspace/D2L.CodeStyle/pull/683/files#r545201100, this area of code will eventually also want to raise a diagnostic for using the wrong audit type as well. While the chances are _slim_ that the following code would occur, we'd want to be able to provide _both_ diagnostics. With that in mind, it may be a better API to accept `diagnosticSink` (`internal delegate void DiagnosticSink( Diagnostic diagnostic );` / `Action<Diagnostic>`) so that we can raise diagnostics as needed internally. `IsAudited( ISymbol symbol, DiagnosticSink diagnosticSink, out Location location )`. With that in mind, it may also make sense to rename it to `CheckAudited`.

```csharp
[Statics.Unaudited( Because.Foo )]
[Mutability.Unaudited( Because.Foo )]
[Mutability.Audited( "Owen Smith", "2020-12-17", "Foo" )]
private bool m_disposed;
```